provider/resource_instance.go

```diff
@@ -90,37 +91,73 @@ func resourceInstance() *schema.Resource {
 					},
 				},
 			},
-			"networks": {
+			"network": {
 				Type:     schema.TypeList,
-				Required: true,
-				ForceNew: true,
-				Elem: &schema.Schema{
-					Type: schema.TypeString,
-				},
-			},
-			"interfaces": {
-				Type:     schema.TypeList,
-				Computed: true,
-				Elem: &schema.Schema{
-					Type: schema.TypeString,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"network_uuid": {
+							Type:        schema.TypeString,
+							Required:    true,
+							ForceNew:    true,
+							Description: "The UUID of the network",
+						},
+						"ipv4": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							Computed: true,
+							Description: "The " +
```

Comment:
Just checking that "optional" and "computed" means I can specify it but if I don't its computed for me?

Reply:
Correct.
*I was quite happy getting all those options for a parameter.*